webapp-django/crashstats/signature/views.py

```diff
@@ -502,22 +502,30 @@ def signature_summary(request, params):
 
     # Augment graphics adapter with data from another service.
     if 'adapter_vendor_id' in facets:
-        vendor_hexes = set()
-        adapter_hexes = set()
+        vendor_hexes = []
+        adapter_hexes = []
         for vendor in facets['adapter_vendor_id']:
-            vendor_hexes.add(vendor['term'])
             for adapter in vendor['facets']['adapter_device_id']:
-                adapter_hexes.add(adapter['term'])
+                vendor_hexes.append(vendor['term'])
+                adapter_hexes.append(adapter['term'])
+
         devices_api = models.GraphicsDevices()
-        try:
-            devices_results = devices_api.get(
-                vendor_hex=vendor_hexes,
-                adapter_hex=adapter_hexes,
+
+        devices_results = {
+            'hits': [],
+            'total': 0,
+        }
+        # In order to avoid hitting the maximum URL size limit, we split the
+        # query in smaller chunks, and then we reconstruct the results.
+        for i in range(0, len(vendor_hexes), 50):
```

Comment:
I did a some checking. 50 of these amounts to a query string that is 1843 bytes, given a random set, as a single query string. That's quite tiny. 

So let's say that's 37 bytes per vendor/hex combo. 

Nginx by default, can read 4 buffers of 4000 bytes. I think. That means, if you allow for some other headers too, we should be able to send about 3x4000 bytes which would mean roughly we should be able to send 324 vendor/adaptor hex combos per request from our webhead to our middleware server.